File: backend-express.js-sockets.io/models/group.js
const mongoose = require("mongoose");
const { Schema } = mongoose;
const User = require("./user");

const groupSchema = new Schema(
    {
        name: {
            type: String,
        },
        members: [
            {
                type: Schema.Types.ObjectId,
                ref: "User",
            },
        ],
        messages: [
            {
                type: Schema.Types.ObjectId,
                ref: "Message",
            },
        ],
    },
    {
        timestamps: true,
    }
);

module.exports = mongoose.model("Group", groupSchema);

File: AndroidApp/vibez/lib/models/core/group.dart
import 'package:hive/hive.dart';
import 'package:vibez/models/core/message.dart';
import 'package:vibez/models/core/user.dart';

part 'group.g.dart';

// flutter pub run build_runner build
@HiveType(typeId: 2)
class Group {
  @HiveField(0)
  String name;

  @HiveField(1)
  List<User> members;

  @HiveField(2)
  List<Message> messages;
}